### ISSUE TITLE
Backport Contour CVE-2021-32783 mitigations

### ISSUE THREAD

**Issue by dprotaso** [projectcontour/contour#3942]
Regarding the ExternalName CVE (https://github.com/projectcontour/contour/security/advisories/GHSA-5ph6-qq5x-7jwc) it would be great if some of the mitigations could be back ported to earlier contour releases.

ie. https://github.com/projectcontour/contour/pull/3934

For example Knative ships an opinionated contour setup. We'd like to mitigate this CVE but not require folks jump N minor versions.

**Comment by youngnick** [comment]
This seems pretty reasonable, since it's a win for pretty much any Contour version. Do you have any specific versions you would like @dprotaso ?

**Comment by dprotaso** [comment]
Knative supports N-4 releases and at this time our oldest supported release uses Contour 1.13. Though that'll be EOL in 6 weeks and the second oldest release uses Contour 1.14

**Comment by youngnick** [comment]
So, we probably need to backport to 1.14, 1.15, 1.16? Since 1.17 and 1.18 already have fixed versions? I think the change is relatively self-contained, so hopefully a cherry-pick will apply cleanly. @xaleeks, thoughts?

**Comment by dprotaso** [comment]
I'd say this is related - https://github.com/projectcontour/contour/issues/3950

We should figure out what exact mitigation to backport. At a minimum the admin socket PR makes sense.

**Comment by xaleeks** [comment]
We had a discussion with the OP and I agree we should backport to v1.14 if we can do a clean cherry pick like you said.  If there are larger changes needed, we can re-evaluate.   @youngnick

**Comment by youngnick** [comment]
Yes, I think that we should backport the domain socket PR back to the 1.14 release if possible.

**Comment by dprotaso** [comment]
Hey I saw https://github.com/projectcontour/contour/pull/3934 merged and was wondering what timelines you have for backporting and creating the patch releases

cc @stevesloka

**Comment by stevesloka** [comment]
Good timing! I just put up a PR that needs some smoke tests to validate: https://github.com/projectcontour/contour/pull/3964

This is for v1.14 right now, once this is good can work through the other releases.
